File: string/_EASY_28.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:

        if len(haystack) < len(needle):
            return -1

        for i in range(len(haystack)):
            if haystack[i : i + len(needle)] == needle:
                return i

        return -1


def main():
    haystack = "hello"
    needle = "ll"
    solution = Solution()
    result = solution.strStr(haystack, needle)
    print(result)


if __name__ == "__main__":
    main()

File: string/_EASY_28.js
var strStr = function(haystack, needle) {
    if (haystack.length < needle.length) {
        return -1;
    }

    for (let i = 0; i <= haystack.length - needle.length; i++) {
        if (haystack.substring(i, i + needle.length) === needle) {
            return i;
        }
    }

    return -1;
};

function main() {
    let haystack = "hello";
    let needle = "ll";
    let result = strStr(haystack, needle);
    console.log(result);
}

main()